qiime2/sdk/action.py

```diff
@@ -283,45 +284,54 @@ def _build_annotations(self):
         return annotations
 
     def _build_numpydoc(self):
-        numpydoc = ""
-        numpydoc += "{}\n\n".format(self.name)
-        numpydoc += "{}\n\n".format(self.description)
+        numpydoc = []
+        numpydoc.append(textwrap.fill(self.name, width=79))
+        numpydoc.append(textwrap.fill(self.description, width=79))
 
         sig = self.signature
         params = True if len(sig.inputs) != 0 else False
 
-        numpydoc += "{}".format(self._build_section(
-            "Parameters", sig.inputs))
-        numpydoc += "{}".format(self._build_section(
-            "Parameters", sig.parameters, parameters=params))
-        numpydoc += ("\n" if len(sig.parameters) != 0 or
-                     len(sig.inputs) != 0 else "")
-        numpydoc += "{}\n".format(self._build_section(
-            "Returns", sig.outputs))
+        inputs = self._build_section("Parameters", sig.inputs)
+        parameters = self._build_section("Parameters", sig.parameters,
+                                         parameters=params)
+        returns = self._build_section("Returns", sig.outputs)
 
-        return numpydoc
+        if len(inputs) != 0 and len(parameters) != 0:
+            parameters = '\n'.join([inputs, parameters])
+            inputs = None
+
+        if inputs is not None:
```

Comment:
The flow control between this block and the above `if` is not terribly clear, it's very much like a state-machine. You might have better luck merging `sig.inputs` and `sig.parameters` (they are both `OrderedDict` objects) and providing that to `_build_section`. Then the presence/absence of them isn't as interesting because you'll either have an empty section or you wont.